### DISCUSSION TITLE
New Pull Request shows Past Commits

### DISCUSSION THREAD

**Discussion by ghost** [community/community#22501] [Pull Requests]
<p>Whenever I try to make a New Pull Request it shows commits from the past I don’t know why , but here’s the example in the image<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/2/8/281c698b65e33e8d03931d4a29b796dbd1df6b32.jpeg" data-download-href="/uploads/short-url/5IPYfO3RUHvXE1eZWKIj3eed3BU.jpeg?dl=1" title="github support" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/74557588/181071507-e9ad9148-0db3-47fb-8da1-57e0a5505deb.jpeg" alt="github support" data-base62-sha1="5IPYfO3RUHvXE1eZWKIj3eed3BU" width="690" height="338" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/2/8/281c698b65e33e8d03931d4a29b796dbd1df6b32_2_690x338.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/2/8/281c698b65e33e8d03931d4a29b796dbd1df6b32_2_1035x507.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/3X/2/8/281c698b65e33e8d03931d4a29b796dbd1df6b32.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/8/281c698b65e33e8d03931d4a29b796dbd1df6b32_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">github support</span><span class="informations">1280×628 76.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>I don’t know the reason why this is happening even if my master branch is up to date with remote repository and the branch that I created from the master is also up to date<br>
<em><strong>Can please someone help me?</strong></em></p>

**Comment by jfullstackdev** [comment]
<p>you want to make a pull request at the original repo ?</p>
<p>is that repo you are referring to is a fork ?</p>
<p>if you will make a new pull request, all the changes you’ve made from the past at your forked repo is included,</p>
<p>to avoid this, delete the fork and fork the repo again, make the changes you only intend for that pull request and never do anything other than that and create a new pull request</p>

**Comment by ghost** [comment]
<p>Yes, the repository is a fork but is there any other way😥?<br>
I just don’t want to delete this repository and fork again.</p>

**Comment by airtower-luna** [comment]
<p>There’s no need to delete the fork. Just create a new branch starting from the upstream default branch, and commit your changes there (e.g. by cherry-picking just the commits you want).</p>
<p>That said, if you want to contribute more in the future, you might want to separate changes you want and don’t want to contribute on separate branches. Then you can rebase or merge into the changes that aren’t for contributions to keep them up to date.</p>

**Comment by jfullstackdev** [comment]
<aside class="quote no-group" data-username="KuldeepBorkar" data-post="3" data-topic="230651">
<div class="title">
<div class="quote-controls"></div>
 KuldeepBorkar:</div>
<blockquote>
<p>I just don’t want to delete this repository and fork again.</p>
</blockquote>
</aside>
<p>yup, that’s correct, if you want to avoid deleting it, create a new branch,</p>
<p>back then, when there was no sync button on Web UI, so that was my tendency, simply delete and fork the repo again,</p>

**Comment by ghost** [comment]
<p>Thank you for the response @airtower-luna😄</p>
<p>Can you please tell the commands;</p>
<p> → How to create a branch from the upstream default branch.</p>
<p><em>That said, if you want to contribute more in the future, you might want to separate changes you want and don’t want to contribute on separate branches. Then you can rebase or merge into the changes that aren’t for contributions to keep them up to date.</em></p>
<p>I don’t really know for sure I did understand the above the statement properly can you please elaborate a bit I am a rookie🥲</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="KuldeepBorkar" data-post="6" data-topic="230651">
<div class="title">
<div class="quote-controls"></div>
 KuldeepBorkar:</div>
<blockquote>
<p>→ How to create a branch from the upstream default branch.</p>
</blockquote>
</aside>
<p>You need to have a local repository, with both your fork and the upstream repository as configured as remotes. If the upstream repository is known as <code>upstream</code> you can do something like this to create and check out a new branch starting from the <code>main</code> branch in the upstream repository:</p>
<pre><code class="lang-auto">git checkout -b my-new-feature upstream/main
</code></pre>
<p>Implement your feature on the branch, and push it to your fork (I’m assuming it’s the remote called <code>origin</code>):</p>
<pre><code class="lang-auto">git push -u origin my-new-feature
</code></pre>
<p>And then you can create your PR. If the concept of <em>remotes</em> (remote repositories) isn’t very familiar to you yet, check: <a href="https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes" class="inline-onebox">Git - Working with Remotes</a></p>
<aside class="quote no-group" data-username="KuldeepBorkar" data-post="6" data-topic="230651">
<div class="title">
<div class="quote-controls"></div>
 KuldeepBorkar:</div>
<blockquote>
<p>I don’t really know for sure I did understand the above the statement properly can you please elaborate a bit I am a rookie🥲</p>
</blockquote>
</aside>
<p>It sounds to me like you have some changes (commits) that you want to contribute to upstream, and some adjustments for yourself. If you keep those on separate branches, I assume you eventually want to have the new features in your adjustment branches, too. Merging and rebasing are the two common ways to do that.</p>
<p>I highly recommend you read all of chapter 3 (“Git Branching”) of the <a href="https://git-scm.com/book/en/v2" class="inline-onebox">Git - Book</a>, but the most important subchapters are <a href="https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging" class="inline-onebox">Git - Basic Branching and Merging</a> and <a href="https://git-scm.com/book/en/v2/Git-Branching-Rebasing" class="inline-onebox">Git - Rebasing</a>. Actually you might want to start with chapter 1 or 2 to help get you familiar with Git, depending on what you know already. 2 and 3 are the most important chapters of the book. :smiley_cat:</p>
